### DISCUSSION TITLE
Support group assignment from token claims

### DISCUSSION THREAD

**Discussion by EternalDeiwos** [outline/outline#2206] [Ideas]
As mentioned in #2205, it would be really great to automate granting privileges based on the contents of token claims. For any deployment spanning more than a handful of people it becomes cumbersome and a security problem waiting to happen to have to maintain privileges in multiple places.

Two separate problems exist here:

1. Reading token claims and assigning users to groups.
2. Updating (synchronizing) group assignments; i.e. if someone is added or removed from a group at the IdP level then it should be reflected in Outline.

**Comment by kristianvld** [comment]
It would also be great to be able to assign the `admin` property through OIDC claims. It could be it's own claim, but I believe it would make more sense to add it into a groups/roles array claim along with group assignment, as this is how admin permissions are managed in most IAM solutions. This way there would also not be a need to add another claim. 

Alternatively, this could also be moved to the group level in the UI, making it so that being member of certain groups makes all members of that group admins.

**Reply by EternalDeiwos** [reply]
@kristianvld I do like the idea of having all members of a particular group being designated as admin as that would be appropriate for RBAC. I think making something like that available via UI and/or via env configuration would be great, but possibly as a future extension. I'd say that having group/role assignment for typical users would be a good intermediate milestone toward that.

Right now I have an open draft PR (#2568) for the above that is awaiting some feedback.

**Reply by kristianvld** [reply]
Oh dang, nice! Looking forward to seeing this merged into production!
